### ISSUE TITLE
[terra-core] Update Keyboard Focus styles to pass '2.4.11 Focus Visible (Enhanced)'

### ISSUE THREAD

**Issue by neilpfeiffer** [cerner/terra-core#3174]
# Feature Request

## Description

WCAG 2.1 introduced criterion [2.4.11: Focus Visible (Enhanced)](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced) which raises the [2.4.7: Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html) criterion standard of when a user interface element displays a visible keyboard focus that it's color contrast ratio be at least 3:1 or thickness of at least 2 CSS pixels.

All interactive terra elements have a visible keyboard focus style present, but not tested to see if they meet automated AXE testing for the 2.4.7 /  2.4.11 criterion. All components in `tera-core` need to be audited and provide updated styling as needed, as well as adding any appropriate additional tests for confirmation of pass/fail.

\
.
## Reference Context / Screenshots
Example: focus styling for `terra-button` does not meet the color contrast ratio or thickness rule when tested:

<img width="1162" alt="image" src="https://user-images.githubusercontent.com/42231146/93592991-db8a2900-f9d0-11ea-981b-567b93a23f92.png">

## Solution

Audit, Add testing, and UX to provide appropriate changes to all components in terra-core

Components affected:
- [ ] _TODO: list out each component needing update here..._

## Related Issues
_add related issue numbers_
- https://github.com/cerner/terra-core/issues/3168

**Comment by neilpfeiffer** [comment]
Keeping blocked on needing UX + A11Y input. Backlogging as future enhancement.

**Comment by neilpfeiffer** [comment]
Closing a duplicate issue [terra-core #3304 (closed)](https://github.com/cerner/terra-core/issues/3304) and transferring the details to this issue as it's replacement.
